pkg/config/identity/kubernetes_auth_test.go

```diff
@@ -0,0 +1,182 @@
+package identity
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	mock_auth_credentials "github.com/3scale-labs/authorino/pkg/common/auth_credentials/mocks"
+	mock_common "github.com/3scale-labs/authorino/pkg/common/mocks"
+
+	envoy_auth "github.com/envoyproxy/go-control-plane/envoy/service/auth/v3"
+	"github.com/golang/mock/gomock"
+	"gotest.tools/assert"
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	authenticationv1 "k8s.io/client-go/kubernetes/typed/authentication/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/flowcontrol"
+)
+
+type authenticatorClientMock struct{}
+
+func (client *authenticatorClientMock) GetRateLimiter() flowcontrol.RateLimiter {
+	return nil
+}
+
+func (client *authenticatorClientMock) Verb(_ string) *rest.Request {
+	return nil
+}
+
+func (client *authenticatorClientMock) Post() *rest.Request {
+	return nil
+}
+
+func (client *authenticatorClientMock) Put() *rest.Request {
+	return nil
+}
+
+func (client *authenticatorClientMock) Patch(_ types.PatchType) *rest.Request {
+	return nil
+}
+func (client *authenticatorClientMock) Get() *rest.Request {
+	return nil
+}
+
+func (client *authenticatorClientMock) Delete() *rest.Request {
+	return nil
+}
+
+func (client *authenticatorClientMock) APIVersion() schema.GroupVersion {
+	return schema.GroupVersion{}
+}
+
+type tokenReviewData struct {
+	requestToken  string
+	authenticated bool
+}
+
+type tokenReviews struct {
+	tokenReviewData
+}
+
+func (t *tokenReviews) Create(ctx context.Context, tokenReview *authv1.TokenReview, opts metav1.CreateOptions) (*authv1.TokenReview, error) {
+	audiences := []string{"echo-api"}
+
+	return &authv1.TokenReview{
+		Spec: authv1.TokenReviewSpec{
+			Token:     t.requestToken,
+			Audiences: audiences,
+		},
+		Status: authv1.TokenReviewStatus{
+			Authenticated: t.authenticated,
+			User: authv1.UserInfo{
+				Username: "system:serviceaccount:authorino:api-consumer-sa",
+				UID:      "0632367e-2455-44fb-bd01-3cf5b5f4a34b",
+				Groups: []string{
+					"system:serviceaccounts",
+					"system:serviceaccounts:authorino",
+					"system:authenticated",
+				},
+			},
+			Audiences: audiences,
+		},
+	}, nil
+}
+
+type k8sAuthenticationClientMock struct {
+	tokenReviewData
+}
```

Comment:
Another way would be to define a local interface with the exact method we require from the k8s authenticator, initialize it with the real one in the package and then replace it in the tests. It could be done in a following refactor ;)